### ISSUE TITLE
label click opens dropdown, not parent

### ISSUE THREAD

**Issue by flieks** [sohobloo/react-native-modal-dropdown#85]
the label is opening the dropdown
not the container. 
I tried several styles but nothing works. I want to have a wide button with text in the middle (so lots of left/right padding..)

my current code:
```
prefixStyle: {
    flex: 1,
    marginRight: 15,
    height: 35,
    backgroundColor: '#3cb3e6',
    justifyContent: 'center',
    alignItems: 'center',
    width: (Dimensions.get('window').width - 115),
  },
  prefixStyle2: {
    color: 'white',
    backgroundColor: '#3cb3e6',
  },
  prefixStyle3: {
    position: 'absolute',
    left: 100,
    width: (Dimensions.get('window').width - 115),
    justifyContent: 'center',
    alignItems: 'center'
  },
  dropdownTextStyle: {
    flex: 1,
    width: (Dimensions.get('window').width - 115),
    justifyContent: 'center',
    alignItems: 'center',
    textAlign: 'center'
  },
´´´

**Comment by sohobloo** [comment]
I'm not sure about your problem.

**Comment by flieks** [comment]
Sorry here some illustration:
![image](https://user-images.githubusercontent.com/5032389/29669729-b1c0d17e-88e4-11e7-806e-d0610ead35f6.png)
i meant the white label inside the red container is opening the dropdown. 
But it should be the whole red container that triggers opening the dropdown no ? Because text can be small..

**Comment by sohobloo** [comment]
@flieks 
This must be caused by the flex style or component hierarchy. I think it's not hard to solve but I do not have my macbook by my side these days.

**Comment by ricardofavero** [comment]
same problem here
it makes it very difficult to tap

**Comment by flieks** [comment]
indeed, eventually i hacked it by puting an absolute View over it and calling the modal.show()

**Comment by 183959633** [comment]
如果你想手动调用 like this :100: 
add --->ref,

ref={(ModalDrop) => { this.modalDropdown = ModalDrop }}

then,

  _ModalDropdown=()=>{
    this.modalDropdown.show()
  }

**Comment by hklee93** [comment]
Can any of you guys help me with this issue? I don't quite understand the given workaround.

**Comment by hklee93** [comment]
> Can any of you guys help me with this issue? I don't quite understand the given workaround.

I just handled by wrapping ModalDropdown with TouchableOpacity and calling show() on press.

**Comment by a51095** [comment]
what⁉️




| |
杰森
|
|
邮箱：13204117850@163.com
|

签名由 网易邮箱大师 定制

On 10/25/2019 12:42, Hakkyung Lee wrote:

Can any of you guys help me with this issue? I don't quite understand the given workaround.

I just handled by wrapping ModalDropdown with TouchableOpacity and calling show() on press.

**Comment by hklee93** [comment]
> what⁉️ | | 杰森 | | 邮箱：13204117850@163.com | 签名由 网易邮箱大师 定制 On 10/25/2019 12:42, Hakkyung Lee wrote: Can any of you guys help me with this issue? I don't quite understand the given workaround. I just handled by wrapping ModalDropdown with TouchableOpacity and calling show() on press. — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe.

I basically used the same idea. As you mentioned, I added ref to ModalDropdown, and called show()  within TouchableOpacity component. Thanks!

**Comment by a51095** [comment]
有源代码的，非常简单




| |
杰森
|
|
邮箱：13204117850@163.com
|

签名由 网易邮箱大师 定制

On 10/25/2019 13:02, Hakkyung Lee wrote:

what⁉️ | | 杰森 | | 邮箱：13204117850@163.com | 签名由 网易邮箱大师 定制 On 10/25/2019 12:42, Hakkyung Lee wrote: Can any of you guys help me with this issue? I don't quite understand the given workaround. I just handled by wrapping ModalDropdown with TouchableOpacity and calling show() on press. — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe.

I basically used the same idea. As you mentioned, I added ref to ModalDropdown, and called show() within TouchableOpacity component. Thanks!
